src/diffpy/srmise/tests/test_dataclusters.py

```diff
@@ -0,0 +1,140 @@
+import numpy as np
+import pytest
+
+from diffpy.srmise.dataclusters import DataClusters
+
+
+@pytest.mark.parametrize(
+    "inputs, expected",
+    [
+        (
+            {
+                "input_x": np.array([1, 2, 3]),
+                "input_y": np.array([3, 2, 1]),
+                "input_res": 4,
+            },
+            {
+                "x": np.array([]),
+                "y": np.array([]),
+                "data_order": np.array([], dtype=np.int32),
```

Comment:
why are we setting these to int32?

Reply:
yes, I kind of know why you did that, but I am also trying to emphasize a point that tests test behavior, not code, so we don't design tests to make code pass, we design tests to ensure specific behaviors, then we write code to make sure the tests pass.  Then we are sure that the code has the right behavior.  So understand why that is int32 and ask the question, why and is that the best thing?  We may not find the answer and then we can default to mimicking the original code, but we should at least go through that process.

Reply:
Both `data_order` and `clusters` store indices, which should be integers.  (It looks like I also did this with `check_qualidx` in modelcluster.py.)   So I believe this functions as a type hint or perhaps a vestige of habits from strongly typed languages.

Why int32?  I don't remember, but I can make an educated guess.  During original dev I used both 32 bit and 64 bit Python at various times and this might be a (very weak) gesture to compatibility between versions when serializing.  Alternately, sometimes I was getting different results on the same input and, IIRC, found that small numerical differences between the 32 bit and 64 bit versions occasional caused different in branching during the recursive steps of peak extraction.  So I may have specified int32 rather than int as a very early part of those debugging efforts, but never removed it.

I don't think changing this should have an impact -- if I knew about some subtle behavior I almost certainly would have left a comment.  My opinion is that it would be better to complete the conversion to Python3 and test on some full PDFs before seeing what happens if this is changed.  How that impacts what test should be written now, I'll leave to Simon.

Reply:
yes, please remove.  Per @Ainamacar this is technical debt and it generally better to remove it when we find it, than "trying to remember to remove it later if we need to".....